File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-11.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        stack = []
        res = 0
        for i in range(len(heights)):
            while stack and heights[i] <= heights[stack[-1]]:
                h = heights[stack.pop()]
                w = i if not stack else i - stack[-1] - 1
                print(h, w)
                res = max(res, h * w) 
            stack.append(i)
            print(stack)

        
        while stack:
            h = heights[stack.pop()]
            w = len(heights) if not stack else i - stack[-1]
            res = max(res, h * w) 
            print(h, w)
        return res

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-11.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int res{};
        for (int i{}; i < s.size(); ++i) {
            int count{};
            for (int j{i}; j < s.size() ; ++j) {
                if (s[j] != s[i]) count ++;
                if (count > k) break;
                res = std::max(res, j - i + 1);
            }
        }
        return res;
    }
};
